File: src/app/dto/teacher.ts
import { Address } from "./address";
import { Department } from "./department";

export class Teacher {
    id: number;
    firstName: string;
    middleName: string;
    lastName: string;
    email: string;
    title: string;
    employmentDate: Date;
    dateOfBirth: Date;
    address: Address;
    department: Department;
    phoneNumber: string;
}

File: Demo.DTO/Teacher.cs
using System;

namespace Demo.DTO
{
    public sealed class Teacher : IPerson
    {
        public int ID { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Email { get; set; }
        public Department Department { get; set; }
        public string Title { get; set; }
        public DateTime EmploymentDate { get; set; }
        public string MiddleName { get; set; }
        public DateTime DateOfBirth { get; set; }
        public Address Address { get; set; }
        public string PhoneNumber { get; set; }
    }
}
